### PR TITLE
Percy cli

### PR DESCRIPTION
**Pull request by samirans89**
Percy CLI changes for the Percy C# SDK

### CODE REVIEW

File: percy-csharp-selenium/percy-csharp-selenium.csproj [hunk 1]
```diff
@@ -26,5 +26,26 @@
...
+    <Content Include="testapp\bundle.js">
+      <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
+    </Content>
+    <Content Include="testapp\base.js">
+      <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
+    </Content>
+    <Content Include="testapp\base.css">
+      <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
+    </Content>
   </ItemGroup>
```
Comment (samirans89) [inline percy-csharp-selenium/percy-csharp-selenium.csproj:50]: holding off approving until we check what happens when a nuget package is created and imported in the customer project. Does it show these HTML assets in greyed out format? If yes, we would need to figure out another way to copy the assets for GitHub workflows.

File: package.json [hunk 2]
```diff
@@ -0,0 +1,6 @@
+{
+    "private": true,
+    "devDependencies": {
+        "@percy/cli": "^1.0.0-beta.50"
```
Comment (samirans89) [inline package.json]: in case there is a change needed for below separate comment, please update this too: 1.0.0-beta.52

**Comment by Robdel12** [comment]
👋🏼 I'm curious -- can we get clarification why it can't be an instance method on the class (rather than having to init a class)? I remember something being mentioned about parallel suites, which I don't think impacts the SDK. All of our other SDKs follow that pattern with no issues. Curious to see an example of it failing with an instance method, if possible. 

Other than getting the function signature right (`percySnapshot(driver, name, options)`), this looks pretty good for us to take over and make it production ready. 💯 🔥 👏🏼

**Comment by fotinakis** [comment]
So @samirans89, speaking with @Robdel12 the only real comment on this PR is updating the interface to better match what we'd make it in the future in a productionized SDK.

Looking at C# style, I believe this would be:

```csharp
Percy.Snapshot(driver, name, [option-kwargs])
```

example:
```csharp
Percy.Snapshot(driver, "homepage", widths: [1280, 768], percyCss: ".foo { display: none; }")
```

@Robdel12 does this look right?

**Comment by Robdel12** [comment]
That looks pretty close. I did a bit of reading through the [C# spec](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/language-specification/classes) and it looks like we can make this an instance method and we'd use the dictionary (like what's being done now). I wish we could use parameter array's, but they're forced to the same type (and we have a mix of ints & strings). 

So I believe it'd be something like: 

```csharp
Percy.Snapshot(driver, "homepage", new Dictionary<string, object> { { "widths", new List<int> { 768, 992, 1200 } })
```

**Comment by samirans89** [comment]
Hi @Robdel12 and @fotinakis ... thanks for your review comment. 
Our team will be checking w.r.t adding the `driver` in the next 1-1.5 weeks. Would it be fine then to merge this code into main and create a separate PR for that?

@Robdel12 - The current signature for the Dictionary object is same as what you have provided. Let me know if I'm missing anything. Example repo usage: 
https://github.com/samirans89/example-percy-csharp-selenium/pull/3/files#diff-64b6bd4518545437a6793878fd7673f5582cdcffe09b3b6f345209b04d8a4c58R87-R90

https://github.com/samirans89/example-percy-csharp-selenium/pull/3/files#diff-64b6bd4518545437a6793878fd7673f5582cdcffe09b3b6f345209b04d8a4c58L67-R65

**Comment by Robdel12** [comment]
@samirans89 

> Our team will be checking w.r.t adding the driver in the next 1-1.5 weeks. Would it be fine then to merge this code into main and create a separate PR for that?

Totally works! Just as long as there isn't a release cut until then (only because the version is now `1.0.0-alpha` -- don't want to release that from 0.x and then have a breaking change follow in `1.0.0-alpha-1` or whatever is next 😃) 

> The current signature for the Dictionary object is same as what you have provided. _Let me know if I'm missing anything_

Not at all! I was semi-correcting Mike's example (this sadly isn't possible `Percy.Snapshot(driver, "homepage", widths: [1280, 768], percyCss: ".foo { display: none; }")` -- only in Ruby I think can turn that into a hash nicely). All is good here 💯

**Pull request merged**